### PR TITLE
Updates for v1 node client

### PR DESCRIPTION
**Pull request by jhamon**
## Problem

Need to update this example app to demonstrate the upcoming v1 release of the node client. This should not be merged until the v1 release is published and the `package.json` is bumped.

## Type of Change

- [x] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [x] This change requires a documentation update

### CODE REVIEW

File: .env.example
Comment (jhamon) [inline .env.example:3]: I put a sample value for this into the template because when I was going through this project I was initially confused about the purpose of this field.

File: README.md
Comment (jhamon) [inline README.md:117]: No need for these anymore.

File: README.md
Comment (jhamon) [inline README.md:187]: I've decided for now not to create a new `createIndexIfNotExists` function because it's usage is fairly limited while the raw `createIndex` call is quite flexible in the options it accepts. It seems pretty simple to implement the equivalent functionality here and does a better job IMO demonstrating the usage of the client than hiding everything in a util function.

File: README.md
Comment (jhamon) [inline README.md:202]: I can change this number back. I was experimenting with different values here and it seems like changing this number didn't have much impact on overall load time.

File: README.md
Comment (jhamon) [inline README.md]: I need to remove this

File: src/deleteIndex.ts
Comment (jhamon) [inline src/deleteIndex.ts:6]: It doesn't really matter if this validation occurs inside the command function or just after the config is loaded since it's not going to change.

File: src/embeddings.ts
Comment (jhamon) [inline src/embeddings.ts]: This should be a `type` import.

File: src/load.ts
Comment (jhamon) [inline src/load.ts:50]: When `waitUntilReady: true` passed, the promise doesn't resolve until the index has status: ready. It handles the polling for status under the hood.

File: src/pinecone.ts
Comment (jhamon) [inline src/pinecone.ts:10]: There's no longer a need to cache this there's no more async initialization.

File: src/query.ts
Comment (jhamon) [inline src/query.ts:11]: Can remove this since we're checking vars above.

File: src/query.ts
Comment (jhamon) [inline src/query.ts:26]: Namespaces are becoming a paid feature, so I don't think we should require them in order to use this example. But if you did want to target a namespace you would do it like this:

```typescript
const results = await index.namespace('default').query({
    vector: queryEmbedding.values,
    topK,
    includeMetadata: true,
    includeValues: false
})
```

The namespace param is no longer part of the payload. You can also save off a scoped client instance to perform multiple actions on the same namespace

```typescript
const ns = index.namespace('default')
await ns.upsert(...) // upsert in namespace
await ns.query(...) // query in namespace
```

File: tests/integration/semanticSearch.test.ts
Comment (jhamon) [inline tests/integration/semanticSearch.test.ts]: This is probably my favorite simplification compared to the past client 😂

**Comment by jhamon** [comment]
These changes are based on the latest WIP in the ts client. We shouldn't merge them until I publish the release and we can update the package.json file to consume the published updated artifact. But I wanted to share them to show the direction things are heading.

Question:
- Can we lower the required node version to something more people are likely to already have installed? I saw you have a super recent node version (19 something something) required in the `engines` section of the package.json. [Node 18 is LTS](https://nodejs.dev/en/about/releases/), so I think it it makes more sense to pin to this version.

File: .env.example
Comment (austin-denoble) [reply .env.example:3]: Nice, I think I also found this a bit confusing when I went through this example initially, but I was also less aware of all the different terminology at that point.

File: README.md
Comment (austin-denoble) [inline README.md:48]: **tiny nit:**  `"share a common modules, the embedder"` -> `"share a common module, the embedder"`

File: README.md
Comment (austin-denoble) [reply README.md:202]: Seems fine to me. When I ran through this example during the hiring process one of these steps (I believe it was the embedding / upserting) took an extremely long time. I'm not sure if that was something on my end or something with how the example was handling things.

I would have assumed it was this bit here though, yeah: https://github.com/pinecone-io/semantic-search-example/blob/00faff1319d0fab83f87490181284cdf6e5a036b/src/load.ts#L49

**Review by austin-denoble — APPROVED** [review]
Overall the approach here looks good to me, thanks for taking this on. Just make sure to address all those TODO'y comments in the PR before merging.

Approving to unblock ya whenever you're ready to go.

**Pull request merged**
